src/main/java/com/minecolonies/blockout/controls/ItemIcon.java

```diff
@@ -72,19 +72,22 @@ protected void drawSelf(final int mx, final int my)
      */
     private void drawItemStack(final ItemStack stack, final int x, final int y)
     {
+        if (stack == null || stack.isEmpty() || stack.getItem() == null)
+        {
+            return;
+        }
+
         final RenderItem itemRender = mc.getRenderItem();
 
-        GlStateManager.translate(0.0F, 0.0F, GUI_ITEM_Z_TRANSLATE);
-        this.zLevel = GUI_ITEM_Z_LEVEL;
-        itemRender.zLevel = GUI_ITEM_Z_LEVEL;
+        GlStateManager.translate(x, y, GUI_ITEM_Z_TRANSLATE);
+        GlStateManager.scale(this.getWidth() / DEFAULT_ITEMSTACK_SIZE, this.getHeight() / DEFAULT_ITEMSTACK_SIZE, 1f);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a "double" or "BigDecimal" instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2164)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a "double" or "BigDecimal" instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2164)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Upper-case this literal "f" suffix. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS818)